services/api/service.go

```diff
@@ -493,6 +527,84 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(ctx context.Context, opts blockSimOptions) (error, error) {
+	t := time.Now()
+	reqErr, simErr := api.blockSimRateLimiter.Send(ctx, opts.req, opts.isHighPrio, opts.fastTrack)
+	log := opts.log.WithFields(logrus.Fields{
+		"duration":   time.Since(t).Seconds(),
+		"numWaiting": api.blockSimRateLimiter.CurrentCounter(),
+	})
+	if simErr != nil &&
+		simErr.Error() != ErrBlockAlreadyKnown &&
+		simErr.Error() != ErrBlockRequiresReorg &&
+		!strings.Contains(simErr.Error(), ErrMissingTrieNode) {
+		// Mark builder as non-optimistic.
```

Comment:
maybe break this up into a nested if, to make it easier to understand?

```suggestion
	if simErr != nil {
        ignoreError := simErr.Error() == ErrBlockAlreadyKnown || simErr.Error() == ErrBlockRequiresReorg || strings.Contains(simErr.Error(), ErrMissingTrieNode) 
        if !ignoreError{
		    // Mark builder as non-optimistic.
```

Reply:
nice cleanup! done